Library/OcBootManagementLib/DefaultEntryChoice.c

```diff
@@ -197,20 +169,24 @@ InternalDebugBootEnvironment (
 
   for (Predefined = 0; Predefined < 2; ++Predefined) {
     for (Index = 0; Index < BootOrderCount; ++Index) {
-      UefiDevicePath = InternalGetBootOptionData (
-                         BootOrder[Index],
-                         BootGuid,
-                         NULL,
-                         NULL,
-                         NULL
-                         );
+      LoadOption = InternalGetBootOptionData (
+        &LoadOptionSize,
+        BootOrder[Index],
+        BootGuid
+        );
+      if (LoadOption == NULL) {
+        continue;
+      }
+
+      UefiDevicePath = InternalGetBootOptionPath (LoadOption, LoadOptionSize);
       if (UefiDevicePath == NULL) {
         DEBUG ((
           DEBUG_INFO,
           "OCB: %u -> Boot%04x - failed to read\n",
           (UINT32) Index,
           BootOrder[Index]
           ));
+        FreePool (LoadOption);
         continue;
       }
```

Comment:
LoadOption needs to be freed somewhere, no?

Reply:
Ouch, overlooked